### PR TITLE
feat: support embedding model runner

### PR DESCRIPTION
**Pull request by xiaoyi-cheng**
*Issue #, if available:*
Support text embedding model runner.

*Description of changes:*
1. add `embedding` jmes path filed for model runners 
2. if `embedding` field can be extracted from `predict` model output, then return

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/fmeval/model_runners/composers/jumpstart_composer.py [hunk 1]
```diff
@@ -12,17 +14,23 @@ class JumpStartComposer(Composer):
...
         self._model_version = jumpstart_model_version
+        self._is_embedding_model = is_embedding_model
 
     def compose(self, prompt: str) -> JumpStartSerializablePayload:
         """
         Composes the payload for the given JumpStartModel from the provided prompt.
         """
+        # embedding models take raw text as input payload
+        if self._is_embedding_model:
+            return prompt
```
Comment (danielezhu) [inline src/fmeval/model_runners/composers/jumpstart_composer.py:33]: Based on the signature of this method, you should still return an object of type `JumpStartSerializablePayload`. Can you construct such a payload with just the `prompt` data? I would rather stay away from changing the signature to `Union[JumpStartSerializablePayload, str]` and directly returning `prompt` if possible.

File: src/fmeval/model_runners/extractors/json_extractor.py [hunk 2]
```diff
@@ -80,3 +85,24 @@ def extract_output(self, data: Union[List, Dict], num_records: int) -> Union[Lis
...
+        Extract embedding from JSON model output
+
+        :param data: Model response. The embedding_jmespath_expression is used to extract the predicted output. The
+                     predicted output must be a string
+        :param num_records: number of inference records in the model output
+        :return: model output
+        """
+        assert num_records == 1, "JSON extractor does not support batch requests"
+        util.require(
+            self.embedding_jmespath_expression,
```
Comment (danielezhu) [inline src/fmeval/model_runners/extractors/json_extractor.py:100]: Shouldn't this instead be `self.embedding_jmespath`?

File: src/fmeval/model_runners/extractors/jumpstart_extractor.py [hunk 3]
```diff
@@ -33,19 +34,29 @@ class JumpStartExtractor(Extractor):
...
         :param sagemaker_session: Optional. An object of SageMaker session
         """
         self._model_id = jumpstart_model_id
         self._model_version = jumpstart_model_version
         self._sagemaker_session = sagemaker_session if sagemaker_session else get_sagemaker_session()
+        self._is_embedding_model = is_embedding_model
+
+        if self._is_embedding_model:
+            self._embedding_compiler = jmespath.compile(EMBEDDING_JMESPATH_EXPRESSION)
+            return
```
Comment (danielezhu) [inline src/fmeval/model_runners/extractors/jumpstart_extractor.py:59]: Can we guarantee that all Jumpstart embedding models output *only* embeddings and nothing else? By returning early, we're saying that we know for certain that we won't need to set `self._output_jmespath_compiler` and `self._log_prob_compiler`.

**Review by danielezhu — CHANGES_REQUESTED** [review]

File: src/fmeval/model_runners/extractors/jumpstart_extractor.py (hunk 3 above)
Comment (xiaoyi-cheng) [reply src/fmeval/model_runners/extractors/jumpstart_extractor.py:59]: Per offline discussion, the answer is `yes` for now

File: src/fmeval/model_runners/composers/jumpstart_composer.py (hunk 1 above)
Comment (xiaoyi-cheng) [reply src/fmeval/model_runners/composers/jumpstart_composer.py]: Per offline discyssion change to `Union[JumpStartSerializablePayload, bytes]` as won't be able to construct payload.

**Review by danielezhu — APPROVED** [review]

**Review by oyangz — APPROVED** [review]

**Pull request merged**